eclipse-sirius/sirius-web
# [Deck] Add Deck CRUD Operations

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?

- Create a new Deck view model from the provided stereotype
- Import the [test.zip](https://github.com/eclipse-sirius/sirius-web/files/13857934/test.zip)
- Open the Deck representation
- Try to Edit Card
- Try to Add a Card
- Try to remove a card
- A card should be editable once selected. Once selected, press F2 or click to start editing
